payu/models/cesm_cmeps.py

```diff
@@ -156,15 +154,50 @@ def setup(self):
 
         self.runconfig.set("ALLCOMP_attributes", "start_type", start_type)
 
-        # Check pelayout makes sense
-        all_realms = self.realms + ["glc", "lnd"]
+        # check pelayout fits within requested cpucount
         cpucount = int(
             self.expt.config.get('ncpus', multiprocessing.cpu_count())
             )
+        all_realms = self.realms + ["glc", "lnd"]
         for realm in all_realms:
             ntasks = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_ntasks"))
-            assert cpucount >= ntasks, "Insufficient cpus for the pelayout in nuopc.runconfig"
-        
+            rootpe = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_rootpe"))
+            pestride = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_pestride"))
+            # rootpe is zero-based
+            if cpucount < (rootpe + ntasks*pestride):
+                raise ValueError(
+                    f"Insufficient cpus for the {realm} pelayout in nuopc.runconfig"
+                )
+
+        # check iolayout
+        for realm in self.realms:
```

Comment:
```suggestion

```

This has to be removed, since the following check using ncpus for each component, instead of the total ncpus.

Reply:
I refactored the code such that this comment doesn't apply ( I think)